lib/cron_parser.dart

```diff
@@ -148,41 +147,81 @@ class _CronIterator implements HasNext<TZDateTime> {
     _currentDate = _currentDate.add(Duration(minutes: 1));
     while (true) {
-      if (_schedule?.months?.contains(_currentDate.month) == false) {
+      if (_schedule.months?.contains(_currentDate.month) == false) {
         _currentDate = TZDateTime(_currentDate.location, _currentDate.year,
             _currentDate.month + 1, 1);
         continue;
       }
-      if (_schedule?.weekdays?.contains(_currentDate.weekday) == false) {
+      if (_schedule.weekdays?.contains(_currentDate.weekday) == false) {
         _currentDate = TZDateTime(_currentDate.location, _currentDate.year,
             _currentDate.month, _currentDate.day + 1);
         continue;
       }
-      if (_schedule?.days?.contains(_currentDate.day) == false) {
+      if (_schedule.days?.contains(_currentDate.day) == false) {
         _currentDate = TZDateTime(_currentDate.location, _currentDate.year,
             _currentDate.month, _currentDate.day + 1);
         continue;
       }
-      if (_schedule?.hours?.contains(_currentDate.hour) == false) {
+      if (_schedule.hours?.contains(_currentDate.hour) == false) {
         _currentDate = _currentDate.add(Duration(hours: 1));
         _currentDate =
             _currentDate.subtract(Duration(minutes: _currentDate.minute));
         continue;
       }
-      if (_schedule?.minutes?.contains(_currentDate.minute) == false) {
+      if (_schedule.minutes?.contains(_currentDate.minute) == false) {
         _currentDate = _currentDate.add(Duration(minutes: 1));
         continue;
       }
       return _currentDate;
     }
   }
 
+  @override
+  TZDateTime previous() {
+    _currentDate = _currentDate.subtract(Duration(minutes: 1));
+    while (true) {
+      if (_schedule.minutes?.contains(_currentDate.minute) == false) {
+        _currentDate = _currentDate.subtract(Duration(minutes: 1));
+        continue;
+      }
+      if (_schedule.hours?.contains(_currentDate.hour) == false) {
+        _currentDate = _currentDate.subtract(Duration(hours: 1));
+        continue;
+      }
+      if (_schedule.days?.contains(_currentDate.day) == false) {
+        _currentDate = _currentDate.subtract(Duration(days: 1));
+        continue;
+      }
+      if (_schedule.weekdays?.contains(_currentDate.weekday) == false) {
+        _currentDate = TZDateTime(
+          _currentDate.location,
+          _currentDate.year,
+          _currentDate.month,
+          _currentDate.day - 1,
+          _currentDate.hour,
+          _currentDate.minute,
+        );
+        continue;
+      }
+      if (_schedule.months?.contains(_currentDate.month) == false) {
+        _currentDate = TZDateTime(
+          _currentDate.location,
+          _currentDate.year,
+          _currentDate.month - 1,
+          _currentDate.day,
+          _currentDate.hour,
+          _currentDate.minute,
+        );
+        continue;
+      }
+      return _currentDate;
+    }
+  }
+
   @override
   TZDateTime current() {
-    assert(_nextCalled);
```

Comment:
@rbubke Is there any reason why you are checking if `.next()` was called? Because there are now `.next()` and `.previous()`, it makes no sense to me to call first `.next()` and `.previous` before calling `current()`

Reply:
it makes sense if you don't want to extra store the current position.

Reply:
So should we check for `_nextCalled` or `_previousCalled`?
```dart
@override
TZDateTime current() {
  assert(_nextCalled || _previousCalled);
  return _currentDate;
}
```

Reply:
yes

Reply:
Added with 82b42b661e7d2733757301b97ab01909fc1c999e ✅